### PR TITLE
avocado_vt: Move the special name handling to loader

### PR DESCRIPTION
**Pull request by ldoktor**
Avocado-vt requires a special test-name handling. Currently it does that
inside the `loader` and then it drops it only to evaluate it again. This
patch moves the `VirtTest.__init__` name handling into the `loader` and
replaces the non-used handling there.

This should not have any impact on the test name whatsoever as the
original name was not used at all and the loader now creates the name
according to the originally used `VirtTest.__init__` handling, with the
only difference that the test is set earlier in the process.

Signed-off-by: Lukáš Doktor <ldoktor@redhat.com>

### CODE REVIEW

**Comment by ldoktor** [comment]
@clebergnu @adereis this should fix the issue you spotted.

**Comment by ldoktor** [comment]
@Hao-Liu @Andrei-Stepanov could you please verify the test names handling works as expected? Theoretically nothing should change, but for a period of time there were 2 handling of the test name. One was inside test loader, the other during test init time, therefor the name from loaders was completely ignored.

I moved it to loader as we introduced `serialized-test-ids` in avocado and the special name handling inside the test is not allowed anymore (the runner executes test with a given name, it's illogical to change that name on execution).

File: .travis.yml [hunk 1]
```diff
@@ -11,6 +11,7 @@ install:
     - if [ $TRAVIS_PYTHON_VERSION == '2.6' ]; then pip install -r requirements-travis-python26.txt; fi
     - git clone https://github.com/avocado-framework/avocado.git avocado-libs
     - cd avocado-libs
+    - pip install -r requirements-travis.txt
```
Comment (clebergnu) [inline .travis.yml]: OK.

File: avocado_vt/loader.py [hunk 2]
```diff
@@ -153,17 +153,20 @@ def discover(self, url, which_tests=loader.DEFAULT):
...
+            # Evaluate the proper avocado-vt test name
+            test_name = None
+            if self.args.vt_config:
+                test_name = params.get("shortname")
+            elif self.args.vt_type == "spice":
+                short_name_map_file = params.get("_short_name_map_file")
+                if "tests-variants.cfg" in short_name_map_file:
+                    test_name = short_name_map_file["tests-variants.cfg"]
+            if test_name is None:
+                test_name = params.get("_short_name_map_file")["subtests.cfg"]
```
Comment (clebergnu) [inline avocado_vt/loader.py]: Looks good.

**Pull request merged**